src/Compiler/Driver/ReuseTcResults/TcResultsPickle.fs

```diff
@@ -0,0 +1,1745 @@
+﻿module internal FSharp.Compiler.ReuseTcResults.TcResultsPickle
+
+open FSharp.Compiler
+open FSharp.Compiler.AbstractIL.IL
+open FSharp.Compiler.AccessibilityLogic
+open FSharp.Compiler.CheckBasics
+open FSharp.Compiler.CheckDeclarations
+open FSharp.Compiler.CompilerGlobalState
+open FSharp.Compiler.ConstraintSolver
+open FSharp.Compiler.DiagnosticsLogger
+open FSharp.Compiler.NameResolution
+open FSharp.Compiler.ParseAndCheckInputs
+open FSharp.Compiler.Infos
+open FSharp.Compiler.Syntax
+open FSharp.Compiler.TcGlobals
+open FSharp.Compiler.Text.Range
+open FSharp.Compiler.TypedTree
+open FSharp.Compiler.TypedTreeBasics
+open FSharp.Compiler.TypedTreeOps
+open FSharp.Compiler.TypedTreePickle
+open FSharp.Compiler.Xml
+
+open Internal.Utilities
+open Internal.Utilities.Collections
+open Internal.Utilities.Library
+open Internal.Utilities.Library.Extras
+
+
+
+type TcSharedData = { TopAttribs: TopAttribs }
+
+// pickling
+
+let p_stamp = p_int64
+
+let p_stamp_map pv = p_Map p_stamp pv 
+
+let p_non_null_slot f (x: 'a | null) st =
```

Comment:
This is a terrible hack based on top of another terrible hack of even having these slots. This should be eliminated. 
It's current used to artificially "backfill" the bindings, instead they should be also cached and restored normally. 

It's not that easy to do though. That often leads to SO due to recursion here, or too much data stored - and so on.